### PR TITLE
fix: delete non auth user in deleteUserIdFromTenantId

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

(A few sentences about this PR)

## Related issues

- Link to issue1 here
- Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your
changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here
highlighting the necessary changes)

## Checklist for important updates

- [ ] Changelog has been updated
    - [ ] If there are any db schema changes, mention those changes clearly
- [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [ ] Changes to the version if needed
    - In `build.gradle`
- [ ] If added a new paid feature, edit the `getPaidFeatureStats` function in FeatureFlag.java file
- [ ] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them
  in `implementationDependencies.json`.
- [ ] Issue this PR against the latest non released version branch.
    - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the
      latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    - If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR

- [ ] Item1
- [ ] Item2

### CODE REVIEW

File: src/main/java/io/supertokens/multitenancy/Multitenancy.java [hunk 1]
```diff
@@ -328,8 +329,15 @@ public static boolean removeUserIdFromTenant(Main main, TenantIdentifierWithStor
             throw new FeatureNotEnabledException(EE_FEATURES.MULTI_TENANCY);
         }
 
-        return tenantIdentifierWithStorage.getMultitenancyStorageWithTargetStorage()
+        boolean finalDidExist = false;
+        boolean didExist = AuthRecipe.deleteNonAuthRecipeUser(tenantIdentifierWithStorage, userId);
+        finalDidExist = finalDidExist || didExist;
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/multitenancy/Multitenancy.java:334]: the did exit of this doesn't really matter right?

File: src/main/java/io/supertokens/multitenancy/Multitenancy.java [hunk 2]
```diff
@@ -328,8 +329,15 @@ public static boolean removeUserIdFromTenant(Main main, TenantIdentifierWithStor
...
+
+        didExist = tenantIdentifierWithStorage.getMultitenancyStorageWithTargetStorage()
                 .removeUserIdFromTenant(tenantIdentifierWithStorage, userId);
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/multitenancy/Multitenancy.java:337]: Add tests for this PR.

File: src/main/java/io/supertokens/authRecipe/AuthRecipe.java [hunk 3]
```diff
@@ -213,6 +213,34 @@ private static void deleteNonAuthRecipeUser(AppIdentifierWithStorage
         });
     }
 
+    public static boolean deleteNonAuthRecipeUser(TenantIdentifierWithStorage
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/authRecipe/AuthRecipe.java:216]: we need a way to enforce that this function is modified as well whenever the above `deleteNonAuthRecipeUser` function is modified. Either through a test or some code refactor.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/authRecipe/AuthRecipe.java (hunk 3 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/authRecipe/AuthRecipe.java:216]: deleteUserIdFromTenantId test will ensure this.

File: src/main/java/io/supertokens/multitenancy/Multitenancy.java (hunk 2 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/multitenancy/Multitenancy.java:337]: done

**Pull request merged**